server/utils.go

```diff
@@ -57,6 +58,18 @@ func (p *Plugin) KVSet(key string, v interface{}) *model.AppError {
 	return p.API.KVSet(key, data)
 }
 
+func (p *Plugin) isBotDMChannel(channel *model.Channel) bool {
+	if channel.Type != model.CHANNEL_DIRECT {
+		return false
+	}
+
+	if !strings.HasPrefix(channel.Name, p.botUserId+"__") && !strings.HasSuffix(channel.Name, "__"+p.botUserId) {
```

Comment:
I'm quite surprised there isn't a method in `model` for this. But your code is still fine.